samples/python/tutorial_code/video/optical_flow/optical_flow.py

```diff
@@ -33,30 +33,35 @@
 mask = np.zeros_like(old_frame)
 
 while(1):
-    ret,frame = cap.read()
-    frame_gray = cv.cvtColor(frame, cv.COLOR_BGR2GRAY)
-
-    # calculate optical flow
-    p1, st, err = cv.calcOpticalFlowPyrLK(old_gray, frame_gray, p0, None, **lk_params)
-
-    # Select good points
-    if p1 is not None:
-        good_new = p1[st==1]
-        good_old = p0[st==1]
-
-    # draw the tracks
-    for i,(new,old) in enumerate(zip(good_new, good_old)):
-        a,b = new.ravel()
-        c,d = old.ravel()
-        mask = cv.line(mask, (int(a),int(b)),(int(c),int(d)), color[i].tolist(), 2)
-        frame = cv.circle(frame,(int(a),int(b)),5,color[i].tolist(),-1)
-    img = cv.add(frame,mask)
-
-    cv.imshow('frame',img)
-    k = cv.waitKey(30) & 0xff
-    if k == 27:
+    ret, frame = cap.read()
+    if ret:
+        frame_gray = cv.cvtColor(frame, cv.COLOR_BGR2GRAY)
```

Comment:
Please reuse code pattern from already existed in other samples (e.g. [here](https://github.com/opencv/opencv/blob/9bcb006ec7b97a076b289acb6fe7375b4c8d933f/samples/dnn/face_detect.py#L138-L141))

```
ret, frame = cap.read()
if not ret:
    ... message...
    waitKey(0)
    break

... processing code kept here ...
```

Reply:
got it, i have changed it according to the above code example, please check it, thanks